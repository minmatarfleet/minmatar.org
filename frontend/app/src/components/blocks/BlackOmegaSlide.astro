---
import { getLangFromUrl, useTranslations } from '@i18n/utils';

const lang = getLangFromUrl(Astro.url);
const t = useTranslations(lang);

import type { CorporationObject } from '@dtypes/layout_components'

interface Props {
    corporation:            CorporationObject;
    is_user_corporation?:   boolean;
    picture_placement?:     'left' | 'right';
}

const {
    corporation,
    is_user_corporation = false,
    picture_placement = 'left',
} = Astro.props

import CorporationSlide from '@components/blocks/CorporationSlide.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
---

<CorporationSlide
    id="blackomega-slide"
    ceo_image='/images/corporations/omega/ceo.png'
    ceo_character_id={1978535095}
    corporation={corporation}
    is_user_corporation={is_user_corporation}
    picture_placement={picture_placement}
>
    <img slot="ships" class="[ minokawa ]" src="/images/corporations/omega/minokawa.png" />
    <img slot="ships" class="[ vindicator ]" src="/images/corporations/omega/vindicator.png" />

    <Flexblock>
        <p>{corporation.introduction ? corporation.introduction : t('black_omega_description_fallback') }</p>
    </Flexblock>
</CorporationSlide>

<style lang="scss">
    img {
        image-rendering: pixelated;
    }

    .minokawa {
        position: absolute;
        top: 47%;
        left: 60%;
        height: auto;
        width: 30%;
        z-index: 1;
        transform: scale(0);
    }

    .vindicator {
        position: absolute;
        top: 27%;
        left: 17%;
        width: 45%;
        height: auto;
        z-index: 2;
        transform: scale(0);
    }

    @keyframes minokawa {
        from {
            transform: scale(0);
            top: 47%;
            left: 60%;
        }

        to {
            transform: scale(1);
            top: -1%;
            left: 22%;
        }
    }

    @keyframes vindicator {
        from {
            transform: scale(0);
            top: 27%;
            left: 17%;
        }

        to {
            transform: scale(1);
            top: 41%;
            left: 10%;
        }
    }
</style>

<style lang="scss" is:global>
    #blackomega-slide .ceo {
        right: 7%;
    }

    .swiper-slide-active {
        .minokawa {
            animation: minokawa cubic-bezier(0.390, 0.575, 0.565, 1.000) both 0.45s;
            animation-delay: 2s;
            animation-iteration-count: 1;
            transform-origin: top right;
            animation-fill-mode: forwards;
        }
        
        .vindicator {
            animation: vindicator cubic-bezier(0.390, 0.575, 0.565, 1.000) both 0.45s;
            animation-delay: 2.2s;
            animation-iteration-count: 1;
            transform-origin: top right;
            animation-fill-mode: forwards;
        }
    }
</style>