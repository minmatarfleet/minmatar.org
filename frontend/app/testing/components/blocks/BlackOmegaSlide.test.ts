import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import BlackOmegaSlide from "@components/blocks/BlackOmegaSlide.astro";

test("BlackOmegaSlide defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(BlackOmegaSlide, {
    props: {
      corporation: 5,
      is_user_corporation: true,
    },
  });

  expect(result).toMatchSnapshot();
});
