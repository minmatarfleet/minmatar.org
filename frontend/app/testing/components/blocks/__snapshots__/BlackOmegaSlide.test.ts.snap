// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlackOmegaSlide defaults 1`] = `
"<div class="fixed-fluid corporation-intro w-full items-center mirrorwed" id="blackomega-slide" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 650px;--fixed_fluid_breakpoint: 30%;">  <div class="wrapper" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: var(--space-s);--wrapper_padding_inline_end: var(--space-s);--wrapper_padding_block: var(--space-s);--wrapper_max_width: 80rem;">  <div class="flex-block [ content ][ pb-32 lg:pb-0 ] gap---space-xl" data-astro-cid-c2ktugqm="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="!flex !items-center" data-astro-cid-c2ktugqm="true">  <div class="fixed-fluid [ items-center w-full ]" data-astro-cid-c2ktugqm="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture class="[ border ]" data-astro-cid-c2ktugqm> <img loading="lazy" src="https://images.evetech.net/corporations/undefined/logo?size=64" width="64" height="64" alt="undefined corporation logo" data-astro-cid-c2ktugqm> </picture> <h2 data-astro-cid-c2ktugqm></h2>  </div>    </div> <div class="wrapper" data-mod data-astro-cid-yv35cmqx style="--wrapper_padding_inline: 0;--wrapper_padding_inline_end: 0;--wrapper_padding_block: 0;--wrapper_max_width: var(--max-text-width);">      <div class="flex-block gap---space-s" data-astro-cid-74i2vz6w="true" data-mod style="--flex_block_gap: var(--space-s);">  <p data-astro-cid-74i2vz6w>Jump first, ask questions later. A formidable capital force and a passion for Vindicators define the Black Omega lifestyle. We have been leading the universe since the dawn of the capsuleer era and now weâ€™re enjoying mimosas on the beach, basking in the chaos we seeded long ago.</p>  </div>     </div>   <div class="flex-block gap---space-xl" id="corporation-status-undefined" x-data="{
        init() {
            
        }
    }" data-astro-cid-ggpwwpcg="true" data-mod style="--flex_block_gap: var(--space-xl);">  <div class="flex-inline" data-astro-cid-ggpwwpcg="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: flex-start;"> <button x-data="{
            show_corporation_modal() {
                    show_modal({
                        partial: '/partials/corporation_modal_component/?corporation_id=undefined',
                    })
                }
            }" x-on:click.stop.prevent="show_corporation_modal()" data-astro-cid-ggpwwpcg="true" data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" disabled x-init="$el.removeAttribute(&#34;disabled&#34;)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Learn more  </div>   </button>   </div>   <small class="[ status-text ]" data-astro-cid-ggpwwpcg>You are member of this corporation</small> </div>    </div>   </div>   <div class="[ corporation-identity ]" data-astro-cid-c2ktugqm> <picture class="ceo-right" data-astro-cid-c2ktugqm> <button class="[ ceo ]" x-data="{
                    show_character_modal() {
                        show_modal({
                            partial: '/partials/character_modal_component/?character_id=1978535095',
                        })
                    }
                }" x-on:click="show_character_modal()" data-astro-cid-c2ktugqm="true" data-astro-cid-syldxee3="true">  <img src="/images/corporations/omega/ceo.png" height="496" width="940" data-astro-cid-c2ktugqm>  </button>   <img class="[ minokawa ]" src="/images/corporations/omega/minokawa.png" data-astro-cid-74i2vz6w><img class="[ vindicator ]" src="/images/corporations/omega/vindicator.png" data-astro-cid-74i2vz6w> </picture> </div>  </div>      "
`;
